### PR TITLE
Fixes #8756: Generates consumer cert RPM on the Capsule.

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by stbenjam** [comment]
It doesn't look like this explicitly requires httpd package - what brings that in, pulp?  

I think we should maybe add `ensure_packages(['httpd'])` to `certs::katello` - thoughts?

File: manifests/init.pp [hunk 1]
```diff
@@ -342,4 +341,9 @@
     }
 
   }
+
+  class { 'certs::katello':
+    deployment_url  => $capsule::rhsm_url,
+    rhsm_port       => $capsule::rhsm_port
```
Comment (stbenjam) [inline manifests/init.pp]: Defined in params.pp, but missing them in the actual capsule class

File: manifests/init.pp [hunk 2]
```diff
@@ -11,7 +11,7 @@
 #
 # $pulp_master::                    whether the capsule should be identified as a pulp master server
 #
-# $pulp_admin_password::            password for the Pulp admin user. It should be left blank so that a random password is generated
+# $pulp_admin_password::            passowrd for the Pulp admin user.It should be left blank so that random password is generated
```
Comment (stbenjam) [inline manifests/init.pp]: ?

File: manifests/init.pp [hunk 3]
```diff
@@ -266,7 +265,7 @@
     }
   }
 
-  $foreman_proxy = $tftp or $dhcp or $dns or $puppet or $puppetca or $realm or $pulp
+  $foreman_proxy = $tftp or $dhcp or $dns or $puppet or $puppetca or $realm
```
Comment (stbenjam) [inline manifests/init.pp]: This and the change above makes me think you need to rebase your checkout?

**Comment by ehelms** [comment]
Updated. Will require https://github.com/Katello/puppet-certs/pull/43

File: manifests/init.pp [hunk 4]
```diff
@@ -11,7 +11,7 @@
 #
 # $pulp_master::                    whether the capsule should be identified as a pulp master server
 #
-# $pulp_admin_password::            password for the Pulp admin user. It should be left blank so that a random password is generated
+# $pulp_admin_password::            password for the Pulp admin user.It should be left blank so that random password is generated
```
Comment (stbenjam) [inline manifests/init.pp]: Looks like this one needs rebasing too, missing b23ca86

File: manifests/init.pp [hunk 5]
```diff
@@ -252,7 +258,6 @@
     } ~>
     class { 'puppet':
       server                      => true,
-      server_ca                   => $puppetca,
```
Comment (stbenjam) [inline manifests/init.pp]: Why?

**Comment by stbenjam** [comment]
Tested with the other 2 PR's and it seems to work fine :+1: , just a couple questions here and some minor puppet-lint failures in Travis.

**Comment by ehelms** [comment]
@stbenjam Up0dated

File: manifests/init.pp [hunk 6]
```diff
@@ -11,7 +11,7 @@
 #
 # $pulp_master::                    whether the capsule should be identified as a pulp master server
 #
-# $pulp_admin_password::            password for the Pulp admin user. It should be left blank so that a random password is generated
+# $pulp_admin_password::            password for the Pulp admin user. It should be left blank so that random password is generated
```
Comment (stbenjam) [inline manifests/init.pp]: Nitpicky, but why does your git client hate this line? The original is better

**Comment by stbenjam** [comment]
ACK pending fixing the $pulp_admin_password description

**Pull request merged**